app/src/pages/inside/testCaseLibraryPage/createTestCaseModal/testCaseDetails/testCaseDetails.tsx

```diff
@@ -66,47 +129,47 @@ export const TestCaseDetails = ({
     }
   }, [stepsData]);
 
-  const handleAddStep = (index?: number) => {
-    setSteps((prevState) => {
+  const handleAddStep = useCallback(
+    (index?: number) => {
       const newStep = createEmptyStep();
+      const updatedSteps = isNumber(index)
+        ? [...steps.slice(0, index + 1), newStep, ...steps.slice(index + 1)]
+        : [...steps, newStep];
+
+      setSteps(updatedSteps);
+      syncStepsToForm(updatedSteps);
```

Comment:
Let's create a function for these two call. I they are repeated three times this way